tests/test_on_data/test_format_converters/test_spikeglx_converter.py

```diff
@@ -96,6 +96,30 @@ class TestConverter(NWBConverter):
         expected_session_start_time = datetime(2020, 11, 3, 10, 35, 10).astimezone()
         self.assertNWBFileStructure(nwbfile_path=nwbfile_path, expected_session_start_time=expected_session_start_time)
 
+    def test_single_probe_spikeglx_converter_in_converter_preview(self):
+        class TestConverter(NWBConverter):
+            data_interface_classes = dict(SpikeGLX=SpikeGLXConverterPipe)
+
+        source_data = dict(SpikeGLX=dict(folder_path=str(SPIKEGLX_PATH / "Noise4Sam_g0")))
+        converter = TestConverter(source_data=source_data)
+
+        conversion_options = dict(
```

Comment:
Thanks.
Let me think a bit on this.

One question:
Do you expect the user to know that the conversion options of each of the sub interfaces are set with their stream names? if so, how how would they? Or are they supposed to be handled by automatic tools?

Reply:
All information for specifying the `conversion_options` dictionary is available in the `conversion_options_schema`

So a user has several options:

a) Read the JSON schema returned by the highest-level `converter.get_conversion_options_schema()`. Not the easiest thing to do but this is the intended route (principle applies to source data schema and metadata schema too)

b) If they defined a custom `NWBConverter` to be nested inside another, they already know their own key references by definition

c) If they used a `ConverterPipe` or core-supported subconverter based on a `ConverterPipe`, then they could find the automatically generated key references via minimal syntax `converter[subconverter_key].data_interface_objects.keys()`. We could even expose a convenience retrieval for it.

The GUIDE of course simplifies all this by automating construction based on the schema structure

Reply:
OK, thanks a lot fo bearing with me so far.

Second question:
Can we just pass conversion_options to all the interfaces os a pipe? Here in this case, they will be the same, right? Is there a case where they will differ? 

I know that they can differ in practice, like, one could theoretically want a different writing buffer or progress bar for one interface of a sub-converter but my question is: is there a place where it would be critical that they are separated?

Reply:
Take a look at bruker/miniscope for that style. There, one common option specifies stubbing. Which is mildly convenient for that but therefore the GUIDE does not support in-file progress updates for those converters, whereas it does for SpikeGLX (because we then have control over each TQDM bar for each data stream in the converter)

This is exactly the kind of question I think is fine for you to consider in the long-term; but for now we really just need this bug fixed and released for one of the more popular formats and uses on the GUIDE.

Reply:
OK, then there is no need to generalize yet.

Move the modified `add_to_nwb` in this branch from to the SpikeGLX converter. This will fix your bug and will avoid us to modify one of the most central functions in the libraryfor a specific case. If we find a better way in the future then we only need to break the SipkeGLX converter.

Reply:
Ok, thinking more about it I think that what bruker does is wrong. Overiding add_to_nwb means that some other utilities like setting the backend configurarion are not propagated in `run_conversion`:

https://github.com/catalystneuro/neuroconv/blob/a970fd44456b7508f43822dc5cf23efdb3285699/src/neuroconv/datainterfaces/ophys/brukertiff/brukertiffconverter.py#L98-L122

Let's make the converters behave like converters and take conversion_options as a dict instead of the "kwargs" arguments.

Check out  https://github.com/catalystneuro/neuroconv/pull/987

Any other converter that comes to mind that is overwriting `add_to_nwb`?

Reply:
Comment still stands though @CodyCBakerPhD 

Let's fix this for nwb converters locally and wait for @weiglszonja on the other PR to do a more general solution that involves changing the central functions.

Reply:
What I'm saying is nothing is outright *wrong* with the current approaches - they are both valid choices in their own respects

- a subconverter that exposes granular control over individual interfaces
- a subconverter that aggregates common commands, like `stub_test` and applies them all through a simpler top-level command

I'm saying that choosing between these, or choosing to keep both at the same time, is the decision that is outside of the scope of this PR, can take a longer time, and receive input from the rest of the team (in particular Szonja who put a lot of those together)

This PR makes a minimal modification to fix a particular bug to allow both styles for flexibility since that reflects the actual current state of things (including any custom converters defined outside of NeuroConv)

I'd rather not delay this fix more than needed since it is rather important for the GUIDE and we're looking to do a release for it soon

Reply:
It is not a minimal modification if it modifies `add_to_nwb` of the `NWBConverter`. Modifying the converter that trigger the bug directly is the minimal fix as I suggested. What's wrong with that?

Reply:
What minimal fix to SpikeGLXConverter are you suggesting that satisfies all the other concerns about its broader use (in particular progress updates)?

Reply:
The function that you added here `add_to_nwb`. 

Add it the converters that need it. That's a localized change.

I thought that it was only spikeglx but it sems that ther is more of them.

Reply:
This is the main converter class that needs it.

As demonstrated in the test, this is for an `NWBConverter` that contains a `SpikeGLXConverter`.

There is no fix to the `SpikeGLXConverter` specifically outside of the general standardization of how to style subconverter conversion options, which is a broader question if it should be standardized at all, and if so which standard should be chosen.

I have added the change directly to the GUIDE in https://github.com/NeurodataWithoutBorders/nwb-guide/pull/908 so I suppose this PR is not necessary if you would prefer to close it - just be aware that programmatic usage of NeuroConv without this fix will have the same issue going forward